sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicyOptions.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.http.HttpResponse;
+
+import java.time.temporal.ChronoUnit;
+import java.util.Objects;
+
+import static com.azure.core.util.CoreUtils.isNullOrEmpty;
+
+/**
+ * Configuration options for {@link RetryPolicy}.
+ */
+@Fluent
```

Comment:
This should be `Immutable` not `Fluent`. We don't want these options changing in the middle of a retry.

Reply:
We did Fluent, to allow client to chain the setters. So we do not need to create many variations of Constructor to set the options.  
If  we do Immutable, They could not be updated which make sense in lot of cases.  But is that true for all 100% of cases ?
We have to choose between @Fluent or @Immutable  , we can not do  both ?

Reply:
After speaking to Allan and Srikanta , we are going with immutable approach.